utils/port_forward.sh

```
@@ -0,0 +1,39 @@
+   curl -sf "http://localhost:${LOCAL_PORT}/" >/dev/null 2>&1; then
+  exit 0
+fi
+
+# Start a background port-forward and keep it running after this script exits
+oc port-forward -n "${NAMESPACE}" svc/"${SERVICE_NAME}" "${LOCAL_PORT}:${LOCAL_PORT}" >/dev/null 2>&1 &
+PF_PID=$!
+# Record PID for caller cleanup
+echo "${PF_PID}" > "${PORT_FORWARD_PID_FILE}"
+
+# Wait up to 30s for the port to respond
+for i in $(seq 1 30); do
```

Shellcheck (0.10.0)

[warning] i appears unused. Verify use (or export if used externally).

(SC2034)